### ISSUE TITLE
xenial: gnome-terminal crashes in Libertine

### ISSUE THREAD

**Issue by TartanSpartan** [ubports/ubuntu-touch#808]
- Device: arale
- Channel: RC
- Build: 2018-W30

### Steps to reproduce
Start gnome-terminal (terminal for short) by touching the application on it's scope or via a shell command.

### Expected behavior
The application starts.

### Actual behavior
The application crashes on startup.

### Logfiles and additional information
http://paste.ubuntu.com/p/qBN6Mw52Pt/

The terminal is essential for Linux usage and system administration, so this must be a high priority issue to fix by OTA-5 or preferably by OTA-4.

**Comment by advocatux** [comment]
I don't see why running a terminal app inside Libertine is "is essential for Linux usage and system administration", you have THE terminal app (https://open-store.io/app/com.ubuntu.terminal) working properly in UT ;)

**Comment by TartanSpartan** [comment]
I disagree. Some admin functions for the container are locked down to the Libertine terminal. The --exec command only goes so far with the phablet terminal. I've managed to give some level of superuser access to the Libertine terminal which augments it's utility, and note that certain locations are only accessible with this terminal, not with the phablet one. Furthermore, the write functions of the container mean that a wide range of CLI functions can be installed and leveraged safely; in the programming realm this includes Python and GCC. This is not so with the case of the phablet terminal, as the phablet in general is write-protected and users have to accept risk to override this and install programs via apt. I'm sure other use cases can also support this line of argument. Thus, there is a solid line of reasoning for why (for those who use Libertine) having a functional terminal app in that container is essential for Linux usage and system administration for the container, alongside the default phablet terminal.

**Comment by padraic7a** [comment]
confirmed on my M10 fhd (freiza) build 2018-09-05
Log: http://paste.ubuntu.com/p/cXgRHzKG54

**Comment by Flohack74** [comment]
@TartanSpartan I dont see how a libertine container App can write to places where the click App can´t. The whole system is mounted r/o, if you need to write somewhere, you need to remount /.

**Comment by Arc676** [comment]
Any chance this is related to #22? It's been closed, but not all apps are working. Firefox, for example, crashes on launch (screenshot in the linked issue).

**Comment by balcy** [comment]
I can confirm this on a fresh xenial libertine container on BQ Aquaris 4.5, RC 16.04 W39.

As a workaround I use lxterminal which works.

starting gnome-terminal from lxterminal gives 

** (gnome-terminal.real:331): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Error creating terminal: The name org.gnome.Terminal was not provided by any .service files

**Comment by hummlbach** [comment]
That seems to be confirmed and is maybe a duplicate of #22(?)

**Comment by dobey** [comment]
> @TartanSpartan I dont see how a libertine container App can write to places where the click App can´t. The whole system is mounted r/o, if you need to write somewhere, you need to remount /.

Libertine itself is not confined, and apps running in libertine can write to pretty much anywhere inside the container. You can also configure additional bind mounts, such as an SD card, or the entire phablet home directory, which will let apps basically write to everywhere that it is possible to write on the device, as an unconfined app. I don't think it is about the host root fs, but just dealing with things outside of normal confinement.

**Comment by dobey** [comment]
> That seems to be confirmed and is maybe a duplicate of #22(?)

I don't think so. I don't think we should group all "app doesn't start under libertine" issues as if they are the same single issue. Granted, there are a lot of duplicate issues, but there are certainly multiple unique problems here. Perhaps this could be re-purposed as "single-instance GNOME apps do not start under libertine" or such, though?

**Comment by TartanSpartan** [comment]
Confirmed on turbo.

**Comment by TartanSpartan** [comment]
To clarify about what I mean when I said that this app can write to places where the click app can't. That's not strictly true. But for the click app, the home directory of the Libertine container is confined to a folder in .local/share/libertine-container/user-data/NAMEOFTHECONTAINER/, while the rest of the root filesystem is in .cache/libertine=container/NAMEOFTHECONTAINER/rootfs/. home is listed within the latter, but you can only _navigate_ to the former by writing out that path, you can't tab to home/phablet within the latter (you get a "Permission denied" refusal). Now, of course it is eminently possible to navigate by writing out the path, but when you use the Libertine container's terminal app, you find that you can use conventional, nested, tabbing navigation from / right down to home or wherever else. This is evidently more _convenient_.

Moreover, refer back to the points I made about superuser access and being able to install whichever CLI utilities you wish, the latter of which certainly cannot be (safely, and in accordance with UT best practice/guidelines) leveraged within the click app. Understand that I have been forced to maintain a Vivid container purely for the functional terminal app (but with all the implications about a full container installation with all the dependencies0 **on top of** a new Xenial container with all the above plus all the other programs I'm interested in. Thank goodness for the Pro 5 64GB version's extra space, but still this is clogging it up too much for my tastes. Please, please make the Xenial libertine container's gnome-terminal functional as soon as you can manage. Thanks :)

**Comment by balcy** [comment]
I've just seen that the command `gnome-terminal --app-id dummy.id` successfully opens the gnome-terminal.
So changing the line `Exec=gnome-terminal` to `Exec=gnome-terminal --app-id dummy.id` in the file `/usr/share/applications/gnome-terminal.desktop` (container file system) is a workaround I think.
I'm not sure why that makes a difference, just stumbled upon it here: https://unix.stackexchange.com/questions/201900/run-true-multiple-process-instances-of-gnome-terminal#201915

**Comment by TartanSpartan** [comment]
@balcy's solution works completely. It's a very simple fix but would make the user experience just that little bit better if it could be patched in an OTA as a simple script (force a change to the .desktop file if gnome-terminal is installed). Hopefully this can be looked at for OTA-13?

**Comment by Fuseteam** [comment]
now the question is how this work be addressed, perhaps something in the way libertine installs stuff?

**Comment by TartanSpartan** [comment]
I have since tried to apply @balcy's workaround to other devices, but it no longer works, so this is still very much a bug.
